firebase-firestore/src/test/java/com/google/firebase/firestore/local/SQLiteIndexManagerTest.java

```diff
@@ -143,14 +143,14 @@ public void testStartAfterFilter() {
   }
 
   @Test
-  public void testEndAtFilter() {
+  public void testEndBeforeFilter() {
```

Comment:
I'm a little confused about the `before` in `Bound`. Based on the naming of this test, if `before == true`, `Bound` does not include the `endAt` value? However, it seems like it's flipped for `startAt` and `startAfter`. 

I find a it harder to reason about before/after since it's flipped for whether it's a startAt, or endAt. Is it possible to change it `before` to an inclusive/exclusive model instead?

Reply:
It's super confusing, but it uses the existing backend terminology:

startAt => before is true (cursor starts right before the provided value)
startAfter => before is false (cursor starts right after the provided value)
endBefore => before is false (cursor stops right before the provided value)
endAt => before is true (cursor stops right after the provided value)

I am confused by this myself, hence the test name here is wrong.

Reply:
oh man this still doesn't make too much sense... Still can't wrap my mind around how endBefore has before set to false in order to mean that the cursor stops before??

Is this documented anywhere? optional nice thing you can do is encapsulate the before bound stuff into some tester objects :)

Reply:
Will clean up in a follow up.